### PR TITLE
Update `fromProtocol` to use `ExceptT`

### PR DESCRIPTION
**Pull request by Jimbo4350**
This PR adds more explicit error handling to `fromProtocol` and its callers.
#404

### CODE REVIEW

File: cardano-node/src/Cardano/CLI/Ops.hs [hunk 1]
```diff
@@ -151,6 +170,57 @@ instance Show CliError where
...
+  -> Maybe DelegationCertFile
+  -> Maybe SigningKeyFile
+  -> Update
+  -> Protocol
+  -> (RunNode ByronBlock
+        => Consensus.Protocol ByronBlock
+        -> ExceptT RealPBFTError IO a)
+  -> ExceptT RealPBFTError IO a
+withRealPBFT gHash nId mNumNodes genFile nMagic sigThresh delCertFp sKeyFp update ptcl action = do
+  SomeProtocol p <- firstExceptT FromProtocolError $ fromProtocol
```
Comment (deepfire) [inline cardano-node/src/Cardano/CLI/Ops.hs]: This should be indented so the subsequent lines don't depend on the `SomeProtocol p <- firstExceptT FromProtocolError`'s length.

File: cardano-node/src/Cardano/CLI/Tx.hs [hunk 2]
```diff
@@ -320,7 +289,10 @@ nodeSubmitTx
...
     withRealPBFT gHash (Just nId) mNumCoreNodes genFile nMagic sigThresh delCertFp sKeyFp update ptcl $
       \p@Consensus.ProtocolRealPBFT{} -> do
-        case gentx of
-          ByronTx txid _ -> putStrLn $ sformat ("TxId: "%Crypto.hashHexF) txid
-          _ -> pure ()
-        handleTxSubmission socketFp p topology gentx stdoutTracer
+        _ <- case gentx of
+               ByronTx txid _ -> pure . putTextLn
+                                      $ sformat ("TxId: "%Crypto.hashHexF) txid
+               otherTxType -> left . TransactionTypeNotHandledYet
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/CLI/Tx.hs]: @denisshevchenko do we want to fail here? I'm not sure.

Comment (denisshevchenko) [reply cardano-node/src/Cardano/CLI/Tx.hs]: @Jimbo4350 I'm not sure as well. :-) Probably yes.

Comment (Jimbo4350) [reply cardano-node/src/Cardano/CLI/Tx.hs]: I agree, was just double checking in case there was a reason why you wanted this to fail silently.

**Pull request merged**